packages/headless/src/components/popover/popover.tsx

```diff
@@ -0,0 +1,49 @@
+import {
+  component$,
+  createContext,
+  Slot,
+  useClientEffect$, useContextProvider,
+  useSignal,
+  useStore,
+} from "@builder.io/qwik";
+import { autoUpdate, computePosition, flip, shift, } from "@floating-ui/dom";
+
+// context
+interface ContextProps { isOpen: boolean }
+export const myContext = createContext<ContextProps>('my-context');
+
+// popover component
+interface PopoverProps {
+  alignments?: 'auto' | 'left' | 'top' | '...';
+}
+
+export const Popover = component$((props: PopoverProps) => {
+  const state = useStore({ isOpen: false });
+
+  useContextProvider(myContext, state);
+
+  const wrapper = useSignal<HTMLElement>();
+
+  useClientEffect$(() => {
```

Comment:
We should avoid client effect. There are better  alternatives for it

Reply:
I have tried the following code but it's invoked about 14 times for each Popover instance 😅

```tsx
  useOnDocument('load', $((event)=> {
    console.log('onload',event)
  }))
```

I don't understand if it's a bug (or a feature 😅)  but I cannot use it to initialize the popover with floating-ui.

Any ideas?

Reply:
@gilf I think we need `useClientEffect$` because the query needs to happen *after* the first render occurred